### PR TITLE
Bring back CLI button (and respective metadata)

### PR DESCRIPTION
**Pull request by sourishkrout**
Slightly refactored to better leverage types plus unit & e2e tests.

Relates to https://github.com/stateful/runme/issues/83. Tests will pass once PR (https://github.com/stateful/runme/pull/84) is merged.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -3,61 +3,35 @@ import { NotebookCellKind } from 'vscode'
...
+    }
   }
 
-  export interface Attribute {
-    [key: string]: any
+  export interface Metadata {
+    ['background']?: string
+    ['interactive']?: string
+    ['closeTerminalOnSuccess']?: string
+    ['mimeType']?: string
```
Comment (christian-bromann) [inline src/types.ts:33]:
```suggestion
    background?: string
    interactive?: string
    closeTerminalOnSuccess?: string
    mimeType?: string
```

File: src/extension/provider/background.ts [hunk 2]
```diff
@@ -46,7 +47,8 @@ export class BackgroundTaskProvider implements vscode.NotebookCellStatusBarItemP
 }
 export class StopBackgroundTaskProvider implements vscode.NotebookCellStatusBarItemProvider {
   provideCellStatusBarItems(cell: vscode.NotebookCell): vscode.NotebookCellStatusBarItem | undefined {
-    const isBackground = cell.metadata?.['background'] === 'true'
+    const metadata = getMetadata(cell)
+    const isBackground = metadata?.['background'] === 'true'
```
Comment (christian-bromann) [inline src/extension/provider/background.ts:51]:
```suggestion
    const isBackground = metadata?.background === 'true'
```

File: src/extension/provider/background.ts [hunk 3]
```diff
@@ -29,7 +29,8 @@ export class ShowTerminalProvider implements vscode.NotebookCellStatusBarItemPro
 
 export class BackgroundTaskProvider implements vscode.NotebookCellStatusBarItemProvider {
   provideCellStatusBarItems(cell: vscode.NotebookCell): vscode.NotebookCellStatusBarItem | undefined {
-    const isBackground = cell.metadata?.['background'] === 'true'
+    const metadata = getMetadata(cell)
+    const isBackground = metadata?.['background'] === 'true'
```
Comment (christian-bromann) [inline src/extension/provider/background.ts:33]:
```suggestion
    const isBackground = metadata?.background === 'true'
```

File: src/extension/utils.ts [hunk 4]
```diff
@@ -5,21 +5,28 @@ import cp from 'node:child_process'
...
 
 import executor from './executors'
 import { ENV_STORE, DEFAULT_ENV } from './constants'
 
 const HASH_PREFIX_REGEXP = /^\s*\#\s*/g
 
+export function getMetadata(cell: vscode.NotebookCell) {
+  const metadata: WasmLib.Metadata = cell.metadata
+  return metadata
+}
```
Comment (christian-bromann) [inline src/extension/utils.ts:18]: Is there a need for this utility function? Is this for typing purposes?

File: src/extension/executors/task.ts [hunk 5]
```diff
@@ -75,7 +75,8 @@ async function taskExecutor(
     //   return new ExperimentalTerminal(scriptFile.fsPath, { cwd, env })
     // })
   )
-  const isBackground = exec.cell.metadata?.['background'] === 'true'
+  const metadata = getMetadata(exec.cell)
+  const isBackground = metadata?.['background'] === 'true'
```
Comment (christian-bromann) [inline src/extension/executors/task.ts:79]: Wouldn't this work too?

```suggestion
  const isBackground = exec.cell.metadata?.background === 'true'
```

**Review by christian-bromann — COMMENTED** [review]
Some comments on the need for the `getMetadata` method.

File: src/extension/utils.ts (hunk 4 above)
Comment (sourishkrout) [reply src/extension/utils.ts:18]: Mainly yes, it also came handy for testing. The type-casting in place seemed unnecessary verbose. I'm open to suggestions if you want to push a "refactor" on top of my branch.

File: src/types.ts (hunk 1 above)
Comment (sourishkrout) [reply src/types.ts:33]: Makes sense.

File: src/extension/utils.ts (hunk 4 above)
Comment (christian-bromann) [reply src/extension/utils.ts:18]: Pushed https://github.com/stateful/vscode-runme/pull/95 without running the tests, I am in some meetings and will clean up after

Comment (sourishkrout) [reply src/extension/utils.ts:18]: 👍 sounds good

**Pull request merged**